src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Resolve.java

```diff
@@ -3780,6 +3779,66 @@ Symbol lookupMethod(Env<AttrContext> env, DiagnosticPosition pos, Symbol locatio
         }
     }
 
+    /**
+     * Find a suitable reference to an enclosing 'A.this' such that A is a subclass of the provided class symbol.
+     * Note that this method can match multiple times during the traversal, as multiple enclosing classes can
+     * be a subclasses of the provided class. Therefore, if this method finds a matching 'A.this' that cannot be accessed
```

Comment:
This comment needs updating to reflect the new logic (i.e., stopping now at the first class found).

Reply:
Good catch!